### PR TITLE
ENH: Implement interchange protocol for DatetimeTZDtype

### LINKED ISSUE #54239: BUG: Conversion of datetime64[ns, UTC] to Arrow C format string is not implemented
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

data = pd.DataFrame(
    {
        "ts": pd.date_range("2018-01-01", periods=5, freq="D"),
        "ts_utc": pd.date_range("2018-01-01", periods=5, freq="D").tz_localize("UTC"),
    }
)

dfi = data.__dataframe__()

print(dfi.get_column_by_name("ts").dtype)
print(dfi.get_column_by_name("ts_utc").dtype)

```
```
(<DtypeKind.DATETIME: 22>, 64, 'tsn:', '=')

---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
Cell In[25], line 13
     10 dfi = data.__dataframe__()
     12 print(dfi.get_column_by_name("ts").dtype)
---> 13 print(dfi.get_column_by_name("ts_utc").dtype)

File ~/miniconda3/envs/dev/lib/python3.10/site-packages/pandas/_libs/properties.pyx:36, in pandas._libs.properties.CachedProperty.__get__()

File ~/miniconda3/envs/dev/lib/python3.10/site-packages/pandas/core/interchange/column.py:126, in PandasColumn.dtype(self)
    124     raise NotImplementedError("Non-string object dtypes are not supported yet")
    125 else:
--> 126     return self._dtype_from_pandasdtype(dtype)

File ~/miniconda3/envs/dev/lib/python3.10/site-packages/pandas/core/interchange/column.py:141, in PandasColumn._dtype_from_pandasdtype(self, dtype)
    137 if kind is None:
    138     # Not a NumPy dtype. Check if it's a categorical maybe
    139     raise ValueError(f"Data type {dtype} not supported by interchange protocol")
--> 141 return kind, dtype.itemsize * 8, dtype_to_arrow_c_fmt(dtype), dtype.byteorder

File ~/miniconda3/envs/dev/lib/python3.10/site-packages/pandas/core/interchange/utils.py:90, in dtype_to_arrow_c_fmt(dtype)
     87     resolution = re.findall(r"\[(.*)\]", typing.cast(np.dtype, dtype).str)[0][:1]
     88     return ArrowCTypes.TIMESTAMP.format(resolution=resolution, tz="")
---> 90 raise NotImplementedError(
     91     f"Conversion of {dtype} to Arrow C format string is not implemented."
     92 )

NotImplementedError: Conversion of datetime64[ns, UTC] to Arrow C format string is not implemented.
```

### Issue Description

We're updating the Altair visualization library to extract DataFrame schema information through the DataFrame Interchange Protocol.  With pandas 2.0.3, This process currently fails for timestamps that include a timezone (but works for timestamps without a timezone).

### Expected Behavior

I expect it to be possible to extract data type information through the DataFrame Interchange Protocol for timestamp-with-timezone columns.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 0f437949513225922d851e9581723d82120684a6
python           : 3.10.11.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.5.0
Version          : Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_US.UTF-8

pandas           : 2.0.3
numpy            : 1.24.3
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.7.2
pip              : 23.1.2
Cython           : None
pytest           : 7.3.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.13.2
pandas_datareader: None
bs4              : 4.12.2
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2023.5.0
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 12.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.1
snappy           : None
sqlalchemy       : 2.0.19
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #54239 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/interchange/column.py [hunk 1]
```diff
@@ -269,7 +274,11 @@ def _get_data_buffer(
             DtypeKind.BOOL,
             DtypeKind.DATETIME,
         ):
-            buffer = PandasBuffer(self._col.to_numpy(), allow_copy=self._allow_copy)
+            if self.dtype[0] == DtypeKind.DATETIME and len(self.dtype[2]) > 4:
```
Comment (MarcoGorelli) [inline pandas/core/interchange/column.py:277]: so anything longer than `'tsn:'` will go here (like `tss:US/Pacific`), nice

**Review by MarcoGorelli — APPROVED** [review]
Nice one, looks good to me

There should probably be a test for interchanging from non-pandas tz-aware, such as
```python
pyarrow.Table
ts: timestamp[us, tz=Asia/Kathmandu]
----
ts: [[2019-12-31 18:15:00.000000,null,2020-01-01 18:15:00.000000]]
```
?

Anyway, not a blocker, I can do that separately

File: pandas/core/interchange/column.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/interchange/column.py:279]: Yeah correct. Added a comment to that effect

**Comment by mroeschke** [comment]
> There should probably be a test for interchanging from non-pandas tz-aware, such as

Yes definitely. Would be good to have that as a follow up

**Comment by MarcoGorelli** [comment]
cool, I'll do that tomorrow

**Pull request merged**